onisoyyc/RaaSberry
# Password Protected In-Place File Encryption

## Changes Made
- Implemented in-place file encryption/decryption instead of creating new files
- Added password verification system with hardcoded password "daniel"
- Improved error handling and resource management
- Added dynamic Documents folder path detection using Windows API
- Removed redundant file creation functions

## Technical Details
- Files are now encrypted in-place using the madryga algorithm
- Added `verifyPassword()` function to gate decryption
- Modified `handleFiles()` and `decryptFiles()` to work directly on original files
- Integrated Windows Shell API for cross-system compatibility
- Added proper memory management and file handle cleanup

## Testing
To test this implementation:
1. Run the executable
2. It will encrypt files in Documents folder
3. Enter password "daniel" when prompted
4. Verify files are properly decrypted

## Security Notes
- Current implementation uses a hardcoded password
- This is a basic implementation for demonstration
- In production, would need:
  - Secure password hashing
  - Key derivation function
  - Network-based key verification
  - Timeout mechanisms

## Related Issues
Closes #[issue-number] - Add password protection to encryption system
